packages/core/src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-button-strategy.ts

```diff
@@ -60,6 +71,50 @@ export default class PaypalCommerceButtonStrategy implements CheckoutButtonStrat
         return Promise.resolve();
     }
 
+    private async _handleOnCancel(_data: OnCancelData) {
+        const lineItems = this._getLineItems();
+        const existingConsignments = this._store.getState().consignments.getConsignmentsOrThrow();
+        const { email } = this._store.getState().billingAddress.getBillingAddressOrThrow();
+        const { firstName, lastName, address1 } = existingConsignments?.[0].shippingAddress || {};
+        const shippingAddress = {
+            ...this._shippingData,
+            firstName: firstName !== 'Fake' ? firstName : '',
+            lastName: lastName !== 'Fake' ? lastName: '',
+            address1: address1 !== 'Fake street' ? address1 : '',
+            email: email !== 'fake@fake.fake' ? email : '',
+            address2: '',
+            stateOrProvince: '',
+            stateOrProvinceCode: '',
+            countryCode: this._shippingData?.countryCode || '',
+            postalCode: this._shippingData?.postalCode || '',
+            city: this._shippingData?.city || '',
+            phone: '',
+            company: '',
+            customFields: []
+        };
```

Comment:
`this._shippingData` should be changed to `this._currentShippingAddress`.

It might be better to set default values for `city`, `countryCode` and `postalCode` in `_transformAddress` method, to avoid using `this._currentShippingAddress` after `...this._currentShippingAddress`.

Reply:
fixed

Reply:
Default values will not work, because `this. _currentShippingAddress` is optional, and ts still giving an error because city, countryCode and postalCode can be undefined because of `this. _currentShippingAddress`.